examples/autoignition-mpi.py

```diff
@@ -224,9 +225,13 @@ def my_rhs(t, state):
 
     def my_checkpoint(step, t, dt, state):
         reaction_rates = eos.get_production_rates(state)
-        viz_fields = [("reaction_rates", reaction_rates)]
+        local_cfl = get_inviscid_cfl(discr, eos=eos, dt=current_dt, cv=state)
```

Comment:
Unfortunately, this is potentially tangled up in the driver structure mess that we're currently in. As it stands, it seems that this recomputes the CFL number separate from the value that's actually used for determining the time step, giving rise to potential inconsistency. Is that correct?

Reply:
*MIRGE-Com* has built-in two modes for timestepping:

Constant CFL:  The user specifies the CFL, and DT is calculated at every step to maintain a constant CFL.  We calculate the domain-global DT as `constant_cfl * stablity_limited_timestep`.

Constant DT: This is the mode we've been operating in since we've been waiting on the grudge back-end for the equivalent of `characteristic_lengthscale`.  In this mode, CFL is not constant; it is a dependent variable that changes with wavespeed (among other things in reactive, viscous flow, but currently just wavespeed). 

Calculating, and visualizing the cell-local CFL in constant DT mode as is done in this example, provides the user a way to examine potential "trouble spots" in the simulation.  The cells with high CFL indicate the DT-limiting areas that require particular care in making a stable sim.

Reply:
> Unfortunately, this is potentially tangled up in the driver structure mess that we're currently in. As it stands, it seems that this recomputes the CFL number separate from the value that's actually used for determining the time step, giving rise to potential inconsistency. Is that correct?

What I mean is that in constant dt mode, which this example is, then CFL needs computing only at viz time based on the user's specified constant dt. We only do it here because the user wants to visualize it.  We don't need or use the actual CFL in the simulation, it's only an example of one of the diagnostic things the user wants to do.  The CFL computed is consistent with the cell-local dts. 

If instead we use constant CFL mode, for which there is no current example, we need to compute dt at every step using the user-specified CFL. In that case the user may like to visualize the cell-local dt's to get an idea of dt-limiting areas. The dt that is computed is consistent with the user's specified CFL, and with that actually used in the simulation.

The driver-adjacent support is here:
https://github.com/illinois-ceesd/mirgecom/blob/6ff984d66909a14d628709f83d6decd85a49846b/mirgecom/simutil.py#L65

I believe this should blend right in with the driver structure development, and is already being used in the y1 drivers.

Reply:
OIC, nvm then.